src/java.base/share/classes/jdk/internal/foreign/abi/BindingSpecializer.java

```diff
@@ -675,6 +659,26 @@ private static String valueLayoutConstantFor(Class<?> type) {
         }
     }
 
+    private void emitInvokeStatic(String ownerInternalName, String methodName, String descriptor) {
```

Comment:
Any reason as to why using a String for the owner instead of Class<?> - you want to avoid multiple computations, I guess? Having a class literal in the clients is much more readable IMHO than having a string constant.

That said, maybe we can find a compromise :-)

Why not naming the INTRN constants in a regular way, so that it's clear what they refer to?
e.g.
```
JAVA_LANG_INVOKE_MEMORYLAYOUT
```
or, if we are ok dropping the package, just:

```
MEMORY_LAYOUT
```

E.g. drop the INTRN suffix - that should make things more readable.

Reply:
Wasn't done yet :) These methods now take a class literal, and I've removed the unneeded INTRN constants